### DISCUSSION TITLE
PowerShell for Exchange Online (can't connect)

### DISCUSSION THREAD

**Discussion by jensjakobsen** [Homebrew/discussions#1045] [Getting started]
Hello

I've installed Homebrew /basically followed the guide here: https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-macos?view=powershell-7), but when I try to connect to Exchange Online with the simplest of commands: connect-exchangeonline, I get this error:

New-ExoPSSession: /Users/jensjakobsen/.local/share/powershell/Modules/ExchangeOnlineManagement/2.0.4/netCore/ExchangeOnlineManagement.psm1:475
Line |
 475 |  … PSSession = New-ExoPSSession -ExchangeEnvironmentName $ExchangeEnviro …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Could not load file or assembly 'Microsoft.Identity.Client, Version=4.23.0.0, Culture=neutral, PublicKeyToken=0a613f4dd989e8ae'. Could not
     | find or load a specific file. (0x80131621)

If I then try another way to login, e.g.:
$O365Session = New-PSSession -ConfigurationName Microsoft.Exchange -ConnectionUri https://outlook.office365.com/powershell-liveid -Credential $O365Cred -Authentication Basic -AllowRedirection

I get another error like:
New-PSSession: This parameter set requires WSMan, and no supported WSMan client library was found. WSMan is either not installed or unavailable for this system.

What the devil am I doing wrong?

**Comment by SMillerDev** [comment]
We don't really provide code/usage support for software we package. You'll have to contact the original developers.

**Reply by jensjakobsen** [reply]
Thanks Sean. I believe that’s Microsoft. I was hoping for some pointers :(

**Reply by carlocab** [reply]
We're about as familiar with the software as you are, so I'm afraid we don't have any pointers to give either. (Homebrew ships about ~9000 different software packages, so it's hard to know things about most of them.) Sorry!

**Reply by jensjakobsen** [reply]
Thank you for taking time to explain the structure.
